src/librustc_middle/ty/print/mod.rs

```diff
@@ -194,17 +194,15 @@ pub trait Printer<'tcx>: Sized {
             .params
             .iter()
             .rev()
-            .take_while(|param| {
-                match param.kind {
-                    ty::GenericParamDefKind::Lifetime => false,
-                    ty::GenericParamDefKind::Type { has_default, .. } => {
-                        has_default
-                            && substs[param.index as usize]
-                                == GenericArg::from(
-                                    self.tcx().type_of(param.def_id).subst(self.tcx(), substs),
-                                )
-                    }
-                    ty::GenericParamDefKind::Const => false, // FIXME(const_generics:defaults)
+            .take_while(|param| match param.kind {
+                ty::GenericParamDefKind::Lifetime => false,
+                ty::GenericParamDefKind::Const { has_default }
+                | ty::GenericParamDefKind::Type { has_default, .. } => {
```

Comment:
you can't use `type_of` for `Const` here, you would have to instead compare the value of the const param

Reply:
I think I fixed it, but I'm not quite sure.